src/andromede/simulation/optimization.py

```diff
@@ -825,7 +849,76 @@ def build_problem(
     database.requirements_consistency(network)
 
     opt_context = OptimizationContext(
-        network, database, block, scenarios, border_management
+        network,
+        database,
+        block,
+        scenarios,
+        border_management,
+        build_strategy,
+        risk_strategy,
+        decision_tree_node,
     )
 
-    return OptimizationProblem(problem_name, solver, opt_context, problem_strategy)
+    return OptimizationProblem(problem_name, solver, opt_context)
+
+
+def fusion_problems(
+    masters: List[OptimizationProblem], coupler: OptimizationProblem
+) -> OptimizationProblem:
+    if len(masters) == 1:
+        # Nothing to fusion. Just past down the master
+        return masters[0]
+
+    root_master = coupler
+    root_master.name = "master"
+
+    root_vars: Dict[str, lp.Variable] = dict()
+    root_cstrs: Dict[str, lp.Constraint] = dict()
+    root_objective = root_master.solver.Objective()
+
+    # We stock the coupler's variables to check for
+    # same name variables in the masters
+    for var in root_master.solver.variables():
+        root_vars[var.name()] = var
+
+    for master in masters:
+        context = master.context
+        objective = master.solver.Objective()
+
+        for var in master.solver.variables():
+            # If variable not already in coupler, we add it
+            # Otherwise we update its upper and lower bounds
+            if var.name() not in root_vars:
+                root_var = root_master.solver.NumVar(var.lb(), var.ub(), var.name())
+                root_master.context._solver_variables[var.name()] = SolverVariableInfo(
+                    var.name(),
+                    len(root_master.context._solver_variables),
+                    context._solver_variables[var.name()].is_in_objective,
+                )
+            else:
+                root_var = root_vars[var.name()]
+                root_var.SetLb(var.lb())
+                root_var.SetUb(var.ub())
+                root_master.context._solver_variables[
+                    var.name()
+                ].is_in_objective = context._solver_variables[
+                    var.name()
+                ].is_in_objective
+
+            for cstr in master.solver.constraints():
+                coeff = cstr.GetCoefficient(var)
+                # If variable present in constraint, we add the constraint to root
+                if coeff != 0:
```

Comment:
Warning float comparison, maybe use something like `math.is_close`

Reply:
In this case is indeed a integer 0 if they are not present in the expression.
It is how the API proposes to check